javascript/ql/src/semmle/javascript/NodeJS.qll

```diff
@@ -162,6 +162,8 @@ private predicate isRequire(DataFlow::Node nd) {
   not nd.getFile().getExtension() = "mjs"
   or
   isRequire(nd.getAPredecessor())
+  or
```

Comment:
Do we dare to use `getACall` here? Shouldn't this be on the more syntactic side, and let the above `isRequire(nd.getAPredecessor())` take care of the data flow?

Reply:
`getACall` only handles data-flow for the `require("module").createRequire` property.  
Data-flow for the result of that call is completely handled  by `isRequire(nd.getAPredecessor())`, there is no overlap. 

I could use the below check instead to make it more syntactic (and miss some flow), but I still want to use `DataFlow::moduleMember()` to support various kinds of imports (destructing etc.). 

```CodeQL
exists(CallExpr call |
  call.getCallee() = DataFlow::moduleMember("module", "createRequire").asExpr()
|
  nd.asExpr() = call
)
```

Reply:
Right. My issue is actually with the recursive use of `moduleMember`, sorry for the `getACall` red herring. I am happy to see it merged as is, but I believe we historically have struggled with the performance/polarity of recursive `moduleImport` predicates. Perhaps I am wrong, @asgerf do you remember?